### PR TITLE
Support openssl v2 module

### PR DESCRIPTION
**Pull request by derekparker**

### CODE REVIEW

**Comment by derekparker** [comment]
There's still some cleanup with regard to the patch files, especially the fixup ones, however I know we've discussed applying the patches directly to the tree anyways, so I'm open to that discussion and essentially squashing and applying all the patches directly to the tree.

File: config/versions.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
     "github.com/golang-fips/go": "main",
-    "github.com/golang-fips/openssl-fips": "b175be2ccd46683a51cba60a9a2087b09593317d",
+    "github.com/golang-fips/openssl-fips": "7544e9a0711b109a2c114b021d079263a3d683a1",
```
Comment (ueno) [inline config/versions.json:3]: As you probably noticed already, the key should be changed to "github.com/golang-fips/openssl" as the other script expects.

**Review by ueno — APPROVED** [review]
The bundled patches look good to me.

**Comment by derekparker** [comment]
@ueno after updating the config it appears there are still failures on centos stream 9 and ubi9.

**Comment by ueno** [comment]
In `crypto/rsa`:
```console
--- FAIL: TestSignPKCS1v15 (0.00s)
--- FAIL: TestVerifyPKCS1v15 (0.00s)
```
should be fixed with https://github.com/ueno/go/commit/6640803d938b82efd32da6459b8f1ee53df5a180.
```console
--- FAIL: TestKeyGeneration (1.28s)
--- FAIL: TestDecryptOAEP (0.00s)
```
openssl/v2 module needs to be updated to include https://github.com/golang-fips/openssl/pull/119

In `crypto/x509`:
```console
--- FAIL: TestGoVerify (0.01s)
```
I guess we could only remove the problematic certificates; I can look into it further if needed.
```console
--- FAIL: TestImports (0.93s)
--- FAIL: TestDisableSHA1ForCertOnly (0.00s)
```
should be fixed with https://github.com/ueno/go/commit/662ae4c37854734b17b095a36e42a8bbbcd85338

**Review by ueno — APPROVED** [review]
LGTM!

**Review by dbenoit17 — APPROVED** [review]

**Pull request merged**